spec/rhc/commands/app_spec.rb

```diff
@@ -1002,4 +1002,342 @@
+        run_output.should match(/Deployment of git ref 'master' in progress for application #{deployment_app_name} .../)
+        run_output.should match(/Success/)
+      end
+    end
+
+    context "git ref successfully with custom ssh executable" do
+      ssh_path = '/usr/bin/ssh'
+
+      let(:arguments) {['app', 'deploy', 'master', '--app', deployment_app_name, '--ssh', ssh_path]}
+      it "should succeed" do
+        @instance.should_receive(:check_ssh_executable!).at_least(:once).and_return(ssh_path)
+        @instance.should_receive(:run_with_system_ssh).at_least(:once)
+        @instance.stub(:exec).and_return([0, "success"])
+        run_output.should match(/Deployment of git ref 'master' in progress for application #{deployment_app_name} .../)
+        run_output.should match(/Success/)
+        expect{ run }.to exit_with_code(0)
+      end
+    end
+
+    context "binary file successfully" do
+      before do
+        @rest_app.stub(:deployment_type).and_return('binary')
+        ssh = double(Net::SSH)
+        session = double(Net::SSH::Connection::Session)
+        channel = double(Net::SSH::Connection::Channel)
+        exit_status = double(Net::SSH::Buffer)
+        exit_status.stub(:read_long).and_return(0)
+        Net::SSH.should_receive(:start).exactly(3).times.with('test.domain.com', 'user', :compression=>false).and_yield(session)
+        session.should_receive(:open_channel).exactly(3).times.and_yield(channel)
+        channel.should_receive(:exec).exactly(3).times.with("oo-binary-deploy").and_yield(nil, nil)
+        channel.should_receive(:on_data).exactly(3).times.and_yield(nil, 'foo')
+        channel.should_receive(:on_extended_data).exactly(3).times.and_yield(nil, nil, '')
+        channel.should_receive(:on_close).exactly(3).times.and_yield(nil)
+        channel.should_receive(:on_request).exactly(3).times.with("exit-status").and_yield(nil, exit_status)
+        lines = ''
+        File.open(@targz_filename, 'rb') do |file|
+          file.chunk(1024) do |chunk|
+            lines << chunk
+          end
+        end
+        channel.should_receive(:send_data).exactly(3).times.with(lines)
+        channel.should_receive(:eof!).exactly(3).times
+        session.should_receive(:loop).exactly(3).times
+      end
+      let(:arguments) {['app', 'deploy', @targz_filename, '--app', deployment_app_name]}
+      it "should succeed" do
+        expect{ run }.to exit_with_code(0)
+        run_output.should match(/Deployment of file '#{@targz_filename}' in progress for application #{deployment_app_name} .../)
+        run_output.should match(/Success/)
+      end
+    end
+
+    [URI('http://foo.com/path/to/file/' + deployment_app_name + '.tar.gz'),
+     URI('https://foo.com/path/to/file/' + deployment_app_name + '.tar.gz')].each do |uri|
+      context "url file successfully" do
+        before do
+        @rest_app.stub(:deployment_type).and_return('binary')
+          ssh = double(Net::SSH)
```

Comment:
I just realised that this is code that you are moving from `deployment_spec.rb`, not code that you wrote, so never mind.